src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,504 @@
+#ifndef STAN_MCMC_CHAINSET_HPP
+#define STAN_MCMC_CHAINSET_HPP
+
+#include <stan/io/stan_csv_reader.hpp>
+#include <stan/math/prim.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/quantile.hpp>
```

Comment:
Nit: These two includes shouldn't be necessary if we're also including math/prim above